server/actions/serveFile.js

```diff
@@ -145,7 +151,44 @@ function serveStaticFile(req, res) {
     res.sendStatus(500);
   });
 
-  stream.pipe(res);
+  let chunks = [];
+  let codeString = "";
+  // IF HTML: push file content chunk-wise from stream to array
+  stream.on("data", chunk => {
+    if (html) {
```

Comment:
This would read easier as a single `if-else`.

```js
if (html) {
  stream.on('data', ...);
  stread.on('close', ...);
}
else {
  stream.pipe(res);
}
```

Reply:
:+1: